lib/ansiblelint/rules/__init__.py

```diff
@@ -72,10 +72,7 @@ def matchlines(self, file, text) -> List[MatchError]:
 
     def matchtasks(self, file: str, text: str) -> List[MatchError]:
         matches: List[MatchError] = []
-        if not self.matchtask:
-            return matches
-
-        if file['type'] == 'meta':
+        if not self.matchtask or file['type'] == 'meta':
```

Comment:
I don't like this refactoring. Simple is better than complex.

Reply:
There is a real reason for doing this, otherwise linting would fail as we would exceed mcabe complexity for this function. See https://github.com/ansible/ansible-lint/issues/744

I tried to avoid it and reduce complexity for this function but this brings a long chain of changes, something more intrusive than expected. I plan to still do it and this is why I assigned the bug to me.

I think it would be better to merge this patch first as it fixes a critical issue with 2.10.

Reply:
Okay. Point is that it should be done separately.